### PR TITLE
Fixing SliverOverlapAbsorber & SliverOverlapInjector child property

### LINKED ISSUE #44282: SliverOverlapAbsorber & SliverOverlapInjector have confusing child/sliver api
These slivers have sliver children, currently specified by the property `child`. This should be changed to `sliver` to match Sliver-Wrapping-Sliver conventions and provide an easier to understand API. :)

### PR DESCRIPTION
**Pull request by Piinks**
## Description

These slivers have sliver children, currently specified by the property child. This should be changed to sliver to match Sliver-Wrapping-Sliver conventions and provide an easier to understand API. 

## Related Issues

Fixes #44282 

## Tests

I added the following tests:

Updated relevant tests.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change. [Announcement Here](https://groups.google.com/g/flutter-announce/c/I7dkV_jxRIc)

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by goderbauer — COMMENTED** [review]
Can we turn this into a soft breakage where we will deprecate the old parameter, but keep it around for a little bit to give people time to upgrade?

File: packages/flutter/lib/src/widgets/nested_scroll_view.dart [hunk 1]
```diff
@@ -1331,14 +1331,17 @@ class SliverOverlapAbsorber extends SingleChildRenderObjectWidget {
   /// [SliverOverlapAbsorberHandle].
   ///
   /// The [handle] must not be null.
-  ///
-  /// The [child] must be a sliver.
   const SliverOverlapAbsorber({
     Key key,
     @required this.handle,
+    @Deprecated('The child property is deprecated, use sliver instead')
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/nested_scroll_view.dart]: This could probably just say "use sliver instead".

Comment (goderbauer) [inline packages/flutter/lib/src/widgets/nested_scroll_view.dart]: Also, can we fit the `@Deprecated` and `Widget sliver` on the same line?

Comment (goderbauer) [reply packages/flutter/lib/src/widgets/nested_scroll_view.dart]: Same everywhere else.

Comment (goderbauer) [reply packages/flutter/lib/src/widgets/nested_scroll_view.dart]: Same everywhere else.

**Review by goderbauer — COMMENTED** [review]
Can you link the breaking change announcement in the PR description?

**Review by passsy — APPROVED** [review]

**Review by goderbauer — APPROVED** [review]
LGTM

This will conflict with https://github.com/flutter/flutter/pull/44618.

**Comment by goderbauer** [comment]
Please merge in master before submission. I believe the new deprecation checks will complain.

File: packages/flutter/lib/src/widgets/nested_scroll_view.dart [hunk 2]
```diff
@@ -1486,9 +1499,17 @@ class SliverOverlapInjector extends SingleChildRenderObjectWidget {
   const SliverOverlapInjector({
     Key key,
     @required this.handle,
+    @Deprecated(
+      'Use sliver instead. '
+        'This feature was deprecated after v1.10.16'
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/nested_scroll_view.dart]: nit: indentation.

Comment (Piinks) [inline packages/flutter/lib/src/widgets/nested_scroll_view.dart]:
```suggestion
      'This feature was deprecated after v1.10.16'
```

File: packages/flutter/lib/src/widgets/nested_scroll_view.dart [hunk 3]
```diff
@@ -1486,9 +1499,17 @@ class SliverOverlapInjector extends SingleChildRenderObjectWidget {
   const SliverOverlapInjector({
     Key key,
     @required this.handle,
+    @Deprecated(
+      'Use sliver instead. '
+      'This feature was deprecated after v1.10.16'
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/nested_scroll_view.dart]:
```suggestion
      'This feature was deprecated after v1.10.16.'
```
I think the analyzer would prefer a fully punctuated sentence. 😝

**Pull request merged**